### ISSUE TITLE
Predictions seem wrong when compared to randomForest

### ISSUE THREAD

**Issue by rmflight** [mnwright/ranger#17]
```r
data(iris)
library(randomForest)
```

```
## randomForest 4.6-12
## Type rfNews() to see new features/changes/bug fixes.
```

```r
iris_spec <- as.factor(iris$Species)
iris_dat <- as.matrix(iris[, !(names(iris) %in% "Species")])
```

```r
set.seed(1234)

test_index <- sample(nrow(iris), 10)
train_index <- seq(1, nrow(iris))[-test_index]
```

```r
iris_train <- randomForest(x = iris_dat[train_index, ], y = iris_spec[train_index], keep.forest = TRUE)
iris_pred <- predict(iris_train, iris_dat[test_index, ])

iris_train$confusion
```

```
##            setosa versicolor virginica class.error
## setosa         47          0         0  0.00000000
## versicolor      0         42         3  0.06666667
## virginica       0          4        44  0.08333333
```

```r
table(iris_pred, iris_spec[test_index])
```

```
##             
## iris_pred    setosa versicolor virginica
##   setosa          3          0         0
##   versicolor      0          5         0
##   virginica       0          0         2
```

```r
library(ranger)
```

```
## 
## Attaching package: 'ranger'
## 
## The following object is masked from 'package:randomForest':
## 
##     importance
```

```r
iris_train2 <- ranger(data = iris[train_index, ], dependent.variable.name = "Species", write.forest = TRUE)
iris_pred2 <- predict(iris_train2, dat = iris[test_index, ])

iris_train2$classification.table
```

```
##             true
## predicted    setosa versicolor virginica
##   setosa         47          0         0
##   versicolor      0         41         3
##   virginica       0          4        45
```

```r
table(iris_pred2$predictions, iris_spec[test_index])
```

```
##             
##              setosa versicolor virginica
##   setosa          0          0         0
##   versicolor      3          0         0
##   virginica       0          5         2
```

**Comment by rmflight** [comment]
OK, so it's not `ranger` itself, but rather the way it is called. Called using the `data = data.frame, dependent.variable.name = "x"`, you get the results above. Using the formula interface, it works perfectly.

```r
iris_train3 <- ranger(Species ~ ., data = iris[train_index, ], write.forest = TRUE)
iris_pred3 <- predict(iris_train3, dat = iris[test_index, ])
table(iris_pred3$predictions, iris_spec[test_index])
```

```
##             
##              setosa versicolor virginica
##   setosa          3          0         0
##   versicolor      0          5         0
##   virginica       0          0         2
```

**Comment by mnwright** [comment]
Thank you for the feedback. The bug is already fixed (see issue #6), but the changes are not on CRAN yet.

**Comment by rmflight** [comment]
Thanks. Didn't think to look at the closed issues.
